api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -160,6 +156,30 @@ public void cleanBillingBuffer() {
             });
   }
 
+  private List<DbBillingProjectBufferEntry> findExpiredCreatingEntries(Instant now) {
+    return billingProjectBufferEntryDao
+        .findAllByStatusAndLastStatusChangedTimeLessThan(
+            DbStorageEnums.billingProjectBufferEntryStatusToStorage(BufferEntryStatus.CREATING),
+            new Timestamp(now.minus(CREATING_TIMEOUT).toEpochMilli()))
+        .stream()
+        // Ensure that we've also synchronized the status since the deadline elapsed. This
+        // covers degenerate cases where syncBillingProjectStatus might be backlogged and
+        // hasn't caught up to check all the CREATING projects, or covers periods where
+        // the Firecloud API is unresponsive. Ideally we'd push this logic directly down
+        // into the DAO, but JPQL has poor support for date comparisons. Keep the above
+        // filter as well as a first pass filter to limit what we pull into memory.
+        .filter(entry -> entry.getLastStatusChangedTime() != null)
+        .filter(entry -> entry.getLastSyncRequestTime() != null)
+        .filter(
+            entry ->
+                Duration.between(
+                            entry.getLastStatusChangedTime().toInstant(),
+                            entry.getLastSyncRequestTime().toInstant())
+                        .toMillis()
```

Comment:
nit: if you prefer, [there's also `Duration.compareTo()`](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#compareTo-java.time.Duration-), which I think can be woven directly into the filter somehow without converting both to millis.

Reply:
I had this initially, but decided it was unreadable; directly comparing these values with a >= sign seems much better to me (would be nice if there were another Duration method or util to do this directly).